week-4/implement/card-validator.test.js

```diff
@@ -0,0 +1,128 @@
+// In this project you'll write a script that validates whether or not a credit card number is valid.
+
+// Here are the rules for a valid number:
+
+// - Number must be 16 digits, all of them must be numbers.
+// - You must have at least two different digits represented (all of the digits cannot be the same).
+// - The final digit must be even.
+// - The sum of all the digits must be greater than 16.
+
+function isValidCardNumber(cardNum) {
+  if (
+    cardNum.length !== 16 ||
+    !Number(cardNum) ||
+    cardNum.slice(-1) % 2 !== 0
+  ) {
+    return false;
+  }
+  // Take first char
+  // Check if it is equal all other chars
+  // Count the equal numbers
+  // return false if count is equal to 16
+
+  let index = 0;
+  let firstChar = cardNum[index];
+  let count = 0;
+
+  while (index < cardNum.length) {
+    if (firstChar === cardNum[index]) {
+      count++;
+    }
+    index++;
+  }
+  if (count === 16) {
+    return false;
+  }
```

Comment:
Good work, like the comments as well

Just a small comment (not a telling off or something to be fixed) Loops can be quite bad performance wise and this code isn't the easiest to read, so in the real world I'd be looking hard for shortcuts when I came to refactor. One that springs to mind is
```
let badCardNum = cardNum[0] * 16;
if (cardNum === badCardNum) {
     return false;
```